0000-product.md

```diff
@@ -0,0 +1,504 @@
+similar to Agent’s metadata. Metadata is a Key,Value store of arbitrary data.
+The GS1 Company Prefix will be stored in the metadata under
+“gs1_company_prefix”.
+
+Solving how to properly provide GS1 Company Prefixes to a Pike Organization will
+be solved in a future RFC. This future RFC should integrate the concepts of GLN
+and the smart contract associated with grid gs1 product should be updated to
+reflect that integration.  `
+
+Trade items can include non-material goods, such as services, which will also
+need to be represented within Grid. This is not covered by this RFC.
+
+Some examples of non-material goods: 
+
+- Batch/Lot (LGTIN) or Serialized (SGTIN) 
+- Logistical Cases/Pallets - Higher levels of packaging of the items - some of
+  which are the trade unit between to parties although not necessarily the
+consumer unit. Identified sometimes with a GTIN but often times, due to their
+heterogeneous nature, with another GS1 identifier (SSCC).
+- Assets (returnable or fixed) - Identified with GRAI or GIAI - I presume like
+  services (GSRN) OUT
+
+To expand the product schema to support all GS1 properties as well as keeping it
+all organized, there will need to be some refactoring done at the grid primitive
+level to support lists in the schema.
+
+This implementation does not account for transfer-of-ownership scenarios, and
+should be implemented as a supplemental RFC at a later time. See:
+https://github.com/hyperledger/grid-rfcs/pull/5#discussion_r312211331
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+Starting with capturing a base product allows us to later extend the definition
+with the complexities associated with products in general. We will be able to
+iterate and develop extensible product/schema definitions for each product space
+i.e.  clothing, furniture, etc.
+
+Some details on products that can be used for a future iteration: [GS1 Apparel
+and General
+Merchandise](https://www.gs1us.org/DesktopModules/Bring2mind/DMX/Download.aspx?Command=Core_Download&EntryId=1067&language=en-US&PortalId=0&TabId=134)
+
+### Clothing:
+
+Material content:\ This element is used to indicate the material composition.
+This element is used in conjunction with the percentage.
+
+Material Percentage:\ Net weight percentage of a product material of the first
+main material. The percentages must add up to 100.
+
+Material Percentage:\ Net weight percentage of a product material of the first
+main material. The percentages must add up to 100.
+
+### Furniture:
+
+Number of pieces:\ The total number of separately packaged components comprising
+a single trade item.
+
+Look at section 3.5.7 Packaging component number AI (243)
+
+### Possible payload modifications:
+
+The owner field is not a _HARD_ requirement in in the product_state.proto. The
+owner (organization) could later be derived from the gtin, as we learn more
+about gtins and integrate some kind of lookup functionality. That functionality
```

Comment:
Note for the future: doing this lookup within the smart contract would require maintaining state which was indexed on GTIN company prefix with meta-data about organization id. The current approach does not require this because the client passes the org id, which can be validated efficiently.

Reply:
Good to know.